### ISSUE TITLE
LinkControl: Improve the link preview and Copy link aria labels

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#60901]
### Description

The LinkControl has a couple ARIA labels that can be improved.

1
The link preview can show a 'rich data' preview of the link destination page _or_ the link text and url. That's important information that needs to be understandable for all users. Visually, it could be improved with some _visible_ title to explain what it is. Assuming a design change is not desired (although I'd recommend it) the semantics needs to be improved.

In fact, the whole 'preview' _and the buttons_ are wrapped in a `div` element with no ARIA role and [an aria label that says 'Currently selected'](https://github.com/WordPress/gutenberg/blob/e33cb8882134317f7b722011f5d2f39ecc4cecdc/packages/block-editor/src/components/link-control/link-preview.js#L82).
I'm not sure what the reasoning about using such a label was, but it's less than ideal. What 'Currently selected' means in this context?

For screen reader users, this section of information needs to be explained in a meaningful way. 'Currently selected' doesn' tell users anything useful. 

Screenshot of Safari and VoiceOver announcing this section of content as 'Currently selected'. Note that the screen reader also announces 'group' because it attempts to assign the most generic ARIA role to a labelled `div` element with no explicit role set.

![Screenshot 2024-04-17 at 12 39 43](https://github.com/WordPress/gutenberg/assets/1682452/ac8ef694-7e4c-4ec3-8fa7-3faa79a488c8)

2
The copy button label (and tooltip) should not contain the link URL.
- Usign such a label makes this button hardily usable for speech recognition software users.
- It's inconsistend with other Copy buttons.
- The aria-label should never contain _values_ or _states_. Instead, it is meant to provide an accessible name for the control, to explain what that control does.

Screenshot:

![Screenshot 2024-04-17 at 12 19 28 2](https://github.com/WordPress/gutenberg/assets/1682452/ce61c346-e4ce-48b6-93ac-1ce167241a82)

### Step-by-step reproduction instructions

- Use a screen reader or explore the link preview by using your browser dev tools inspcctor.
- Observe the link preview is wrapped in a div with an aria-label `Currently selected` and it is announced as such.
- Hover or focus the Copy button.
- Observe the tooltip (which is based on the aria-label) contains the URL value.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
For a new, more meaningful, aria-label for the link preview it's worth reminding the link preview can show a few things that are conceptually different.

The fetched information of an external site displayed in a 'rich data' preview, including the site icon:

![fetched](https://github.com/WordPress/gutenberg/assets/1682452/68f98740-3bb7-4807-8c5d-d0c90c34865b)

Same for an internal link.

Or, if the link destination page can't be fetched, it will display a generic icon, the link text, and the link url with the protocol part stripped out:

![not fetched](https://github.com/WordPress/gutenberg/assets/1682452/a9ccda55-49fc-4596-91f1-90d73c141df3)

Or, when the link text is the same as the link url, it _may_ show only the link:

![image](https://github.com/WordPress/gutenberg/assets/1682452/b86c532f-9322-4582-8994-89e501f10fea)

Given this UI can show different things, sometimes it may actually not be a 'preview'. A new, more meaningul, labeling should likely avoid the term 'preview' and use a more generic terminology to commuincate users this is generic information about the destination page.

**Comment by afercia** [comment]
Note: translatable should avoid concatenation. In the case illustrated below, the colon `:` is not translatable.
The WordPress l10n guidelines recomment to always use _fully translatable strings_, including punctuation.

https://github.com/WordPress/gutenberg/blob/e33cb8882134317f7b722011f5d2f39ecc4cecdc/packages/block-editor/src/components/link-control/link-preview.js#L146-L150

**Comment by afercia** [comment]
Additionally: when the preference 'Show button text labels' is enabled, all three buttons still show the tooltips, they just repeat the visible text. That's redundant and should be avoided as it's done for other buttons in the UI.

Screenshot:

![Screenshot 2024-04-19 at 15 22 29](https://github.com/WordPress/gutenberg/assets/1682452/9fd8208f-ea50-4284-a748-3b4f797df4aa)
